tests/unit/test_charm.py

```diff
@@ -409,6 +389,71 @@ def test_tcp_config(self):
         static_config = charm.unit.get_container("traefik").pull(_STATIC_CONFIG_PATH).read()
         assert yaml.safe_load(static_config)["entryPoints"][prefix] == expected_entrypoint
 
+    def setup_forward_auth_relation(self) -> int:
+        relation_id = self.harness.add_relation("forward-auth", "provider")
+        self.harness.add_relation_unit(relation_id, "provider/0")
+        self.harness.update_relation_data(
+            relation_id,
+            "provider",
+            {
+                "decisions_address": "https://oathkeeper.test-model.svc.cluster.local:4456/decisions",
+                "app_names": '["charmed-app"]',
+                "headers": '["X-User"]',
+            },
+        )
+
+        return relation_id
+
+    @patch("charm.KubernetesServicePatch", lambda *_, **__: None)
+    def test_forward_auth_relation_databag(self):
+        self.harness.set_leader(True)
+        self.harness.update_config({"external_hostname": "testhostname"})
+        self.harness.begin_with_initial_hooks()
+
+        provider_info = {
+            "decisions_address": "https://oathkeeper.test-model.svc.cluster.local:4456/decisions",
+            "app_names": ["charmed-app"],
+            "headers": ["X-User"],
+        }
+
+        _ = self.setup_forward_auth_relation()
+
+        self.assertTrue(self.harness.charm.forward_auth.is_ready())
+
+        expected_provider_info = self.harness.charm.forward_auth.get_provider_info()
+
+        assert expected_provider_info.decisions_address == provider_info["decisions_address"]
+        assert expected_provider_info.app_names == provider_info["app_names"]
+        assert expected_provider_info.headers == provider_info["headers"]
```

Comment:
**Question:** Do these asserts belong here or in forwardauth?

It would be useful to assert that the decisions_address is present in the rendered traefik config yaml.

Reply:
This is covered in integration tests